vignettes/fims-path-maturity.Rmd

```diff
@@ -0,0 +1,549 @@
+  Parameter slope;  /**< the width of the curve at the median */
+  ...
+}
+```
+
+We typically will reference the child class from R to specify the formulation of maturity we wish to use in a specific model run of FIMS. In this particular example, we are using the logistic form of maturity. 
+
+```{r image-path-maturity-1, echo = FALSE, message=FALSE, out.width = '85%'}
+knitr::include_graphics("figures/fims-path-maturity-1.png")
+```
+
+All Rcpp interface classes from FIMS define parameters (e.g. `median`, `slope`) using the `Parameter` class defined in [rcpp_interface_base.hpp](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/interface/rcpp/rcpp_objects/rcpp_interface_base.hpp) under [inst/include/interface/rcpp/rcpp_objects](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/interface/rcpp/rcpp_objects):
+
+```{Rcpp, eval = FALSE}
+class Parameter {
+ public:
+  double value; /**< initial value of the parameter*/
+  double min =
+      std::numeric_limits<double>::min(); /**< min value of the parameter*/
+  double max =
+      std::numeric_limits<double>::max(); /**< max value of the parameter*/
+  bool is_random_effect = false;          /**< Is the parameter a random effect
+                                             parameter? Default value is false.*/
+  bool estimated =
+      false; /**< Is the parameter estimated? Default value is false.*/
+  ...
+  }
+```
+The fields from the `Parameter` class that are accessible from R are defined in the 
+[rcpp_interface.hpp](https://github.com/NOAA-FIMS/FIMS/blob/main/inst/include/interface/rcpp/rcpp_interface.hpp) file under [inst/include/interface/rcpp](https://github.com/NOAA-FIMS/FIMS/tree/main/inst/include/interface/rcpp):
+
+```{Rcpp, eval = FALSE}
+ Rcpp::class_<Parameter>("Parameter")
+      .constructor()
+      .constructor<double>()
+      .constructor<Parameter>()
+      .field("value", &Parameter::value)
+      .field("min", &Parameter::min)
+      .field("max", &Parameter::max)
+      .field("is_random_effect", &Parameter::is_random_effect)
+      .field("estimated", &Parameter::estimated);
+```
+
+Each field (i.e. parameter) from `maturity` (i.e. the maturity module we defined in R) will therefore inherit the five fields defined in the Parameter class: `value`, `min`, `max`, `is_random_effect`, `estimated`. 
+
+## Linking the Rcpp interface to the fims namespace.
+
+### What is the `fims` namespace?
+
+A namespace in C++ is similar to what a library is in R.
+
+The core of FIMS is within the `fims` namespace and is a convenient way to differentiate between the part of the C++ code base that is portable, i.e. independent of statistical platform, and the part of the codebase that depends on platforms outside of base C++ (e.g. `{Rcpp}`, R, `{TMB}`). Any code written within:
+```{Rcpp, eval = FALSE}
+namespace fims{
+  
+}
+```
+is considered to be a part of the `fims` namespace. Classes written within the `fims` namespace can be accessed using `fims::`.
```

Comment:
I think fims:: is accessed using C++ code, not R code?

Reply:
Updated to: C++ classes written within the `fims` namespace can be accessed within the C++ code base using `fims::`